### ISSUE TITLE
Total.ParameterRisk & Total.ParameterRisk

### ISSUE THREAD

**Issue by grishaloshadinov** [mages/ChainLadder#1]
My issue is with your R code for MackChainLadder formula in particular with both “Total.ProcessRisk” and “Total.ParameterRisk” elements. They both seem to produce incorrect results. If you'd like detailed explanation of the issue send me an email to grisha_loshadinov@yahoo.co.uk

**Comment by trinostics** [comment]
Please describe your "issue" and provide a reproducible example.

**Comment by grishaloshadinov** [comment]
to explain the issue I've got to attach excel file, word document and pdf file, how can I possibly do that over here? That's why I asked to send me an email and I actually provided my email address

**Comment by LostTriangles** [comment]
Hi, was this issue ever resolved?

I believe I'm running into the same issue when I pull up MackChainLadder commands for Total.ParameterRisk but not Total.ProcessRisk. Running R 3.3.1 and ChainLadderPackage 0.2.2

I'll use the basic RAA data as an example. 

R> MackChainLadder(RAA) should yield Mack.S.E of 

1981: 0
1982: 143
1983: 592
1984: 713
1985: 1,452
1986: 1,995
1987: 2,204
1988: 5,354
1989: 6,332
1990: 24,566

The total Mack.S.E should be Sqrt(Total Parameter Risk^2 + Total Process Risk^2), which I believe also equivalent to the square root of each year's Mack SE squared.

For that value, I'm getting 26,154.79.
In R, I'm seeing in the Totals section at the bottom 26,880.74 instead.

Digging deeper into the issue, I decided to look into the relationship between ProcessRisk/Parameter Risk and TotalProcessRisk/TotalParameterRisk.

The Total ProcessRisk/ParameterRisk for any given development period should simply be the square root of the sum of squares for each origin year at that development period from our Process/Parameter Risk triangles, right? 

In other words:

R> MackChainLadder(RAA)$Total.ProcessRisk should show 12824.272 for dev period 3. 
R> MackChainLadder(RAA)$Mack.ProcessRisk should show 2445.506 and 12588.942 along the period 3 triangle column.

Sqrt(2445.506^2+12588.942^2) = 12824.272... The same logic applies at the other development periods as well. 

However, these calculations don't seem to hold true when querying for ParameterRisk, although I would expect them to be in order to get the 26,154.79 I referenced above. 

R> MackChainLadder(RAA)$Total.ParameterRisk shows 4099.369 at dev period 3. 
R> MackChainLadder(RAA)$Mack.ParameterRisk shows 732.8359 and 3877.6111 along the dev period 3 column.

Square Root(732.8359^2 + 3877.6111^2) = 3946.253 <> 4099.369

Is this a bug in the package? 

Thanks in advance.

**Comment by chiefmurph** [comment]
I apologize for the late reply. 

"The Total ProcessRisk/ParameterRisk for any given development period should simply be the square root of the sum of squares for each origin year at that development period from our Process/Parameter Risk triangles, right?"

No, that is incorrect. Yes, the process risk is assumed to be independent by origin year. But the parameter risk cannot be independent by origin year because the origin years use the same estimated link ratio, and therefore are correlated. So the square root of the sum of the squares does not work for parameter risk of the Total.

To understand the formula for the Total, think of the Total value in a development period as the projection of the previous total (which is an estimate) plus the projection of the value (a scalar) of the "next" origin year that gets into that particular Total. The parameter risk formula of that sum uses the same recursive derivation as the parameter risk formula of any individual origin year, with a bit more complication due to the mix of an estimate and a scalar in the sum. Please refer to my paper "Unbiased Loss Development Factors" for a more thorough explanation.

Again, apologies for the delay, and thank you for your interest in ChainLadder!

Best regards,
Dan Murphy

**Comment by LostTriangles** [comment]
Hi Dan,

No worries at all, I thank you very much for your response! My second guess would have been that there was some type of Covariance coming into play which was not being shown in the summaries. 

The Mack excel handout on the CAS site actually did show a Covariance Matrix for Mack's example in his paper so I must have overlooked that - but everything makes sense now. 

Thanks again for your explanation, and I am definitely appreciative of the ChainLadder!

**Comment by trinostics** [comment]
Not a ChainLadder issue
